src/main/kotlin/org/rust/ide/wordSelection/RsFieldSelectionHandler.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.wordSelection
+
+import com.intellij.codeInsight.editorActions.ExtendWordSelectionHandlerBase
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.TreeUtil
+import org.rust.lang.core.psi.RsElementTypes
+import org.rust.lang.core.psi.RsEnumVariant
+import org.rust.lang.core.psi.RsFieldDecl
+import org.rust.lang.core.psi.RsStructLiteralField
+
+class RsFieldSelectionHandler : ExtendWordSelectionHandlerBase() {
+    override fun canSelect(e: PsiElement): Boolean = isFieldDecl(e)
+
+    override fun select(e: PsiElement, editorText: CharSequence, cursorOffset: Int, editor: Editor): List<TextRange>? {
+        val start = e.textRange.startOffset
+        var end = e.textRange.endOffset
+        TreeUtil.skipWhitespaceAndComments(e.node.treeNext, false)?.let {
+            if (it.elementType == RsElementTypes.COMMA) end = it.textRange.endOffset
+        }
+        return ExtendWordSelectionHandlerBase.expandToWholeLine(editorText, TextRange.create(start, end))
+    }
+
+    companion object {
+        internal fun isFieldDecl(e: PsiElement) =
```

Comment:
1. The name looks confusing a bit. Consider rename it to `isFieldLike` or a something. (@Undin suggestions?)
2. `internal` it not an analogue of java's package-private, we don't use it because we are not a library. It package-private case we have to use (default) public =(
3. Two tabs instead of one

And maybe `RsMatchArm` could be in this list

Reply:
1.  sure, renamed.
2. thanks for the hint.
3. project's auto formatter rules

RsMatchArm - we'll check, thanks